plugins/account_history_plugin/account_history_plugin.cpp

```diff
@@ -0,0 +1,132 @@
+#include <eos/account_history_plugin/account_history_plugin.hpp>
+#include <eos/account_history_plugin/account_history_object.hpp>
+#include <eos/chain/chain_controller.hpp>
+#include <eos/chain/config.hpp>
+#include <eos/chain/exceptions.hpp>
+#include <eos/chain/transaction.hpp>
+#include <eos/chain/types.hpp>
+
+#include <fc/crypto/sha256.hpp>
+#include <fc/io/json.hpp>
+#include <fc/variant.hpp>
+
+#include <boost/multi_index/hashed_index.hpp>
+#include <boost/multi_index/mem_fun.hpp>
+#include <eos/chain/multi_index_includes.hpp>
+
+namespace fc { class variant; }
+
+namespace eos {
+
+using chain::block_id_type;
+using chain::ProcessedTransaction;
+using chain::signed_block;
+using boost::multi_index_container;
+using chain::transaction_id_type;
+using namespace boost::multi_index;
+
+class account_history_plugin_impl {
+public:
+   ProcessedTransaction get_transaction(const chain::transaction_id_type&  transaction_id) const;
+   void applied_block(const signed_block&);
+   chain_plugin* chain_plug;
+private:
+
+   optional<block_id_type> find_block_id(const transaction_id_type& transaction_id) const;
+};
+
+optional<block_id_type> account_history_plugin_impl::find_block_id(const transaction_id_type& transaction_id) const
+{
+   const auto& db = chain_plug->chain().get_database();
+   optional<block_id_type> block_id;
+   db.with_read_lock( [&]() {
```

Comment:
Were you able to get it to work, or is there an issue with my commit?  I had a heck of a time getting the submodule commit correct, so it would not be a big shocker if it really didn't work.